posthog/temporal/data_imports/pipelines/meta_ads/source.py

```diff
@@ -0,0 +1,430 @@
+    schema = get_schemas()[config.resource_name]
+    is_stats = config.resource_name.endswith("_stats")
+
+    def get_rows() -> collections.abc.Iterator[list[dict]]:
+        integration = get_integration(config, team_id)
+        access_token = integration.access_token
+
+        # Determine date range for incremental sync
+        time_range = None
+
+        if should_use_incremental_field:
+            if incremental_field is None or incremental_field_type is None:
+                raise ValueError("incremental_field and incremental_field_type can't be None")
+
+            if db_incremental_field_last_value is None:
+                last_value: dt.date = dt.date.today() - dt.timedelta(days=META_ADS_MAX_HISTORY_DAYS)
+            else:
+                last_value = db_incremental_field_last_value
+
+            start_date = last_value.strftime("%Y-%m-%d")
+            # Meta Ads API is day based so only import if the day is complete
+            end_date = (dt.date.today() - dt.timedelta(days=1)).strftime("%Y-%m-%d")
+            time_range = {
+                "since": start_date,
+                "until": end_date,
+            }
+        else:
+            time_range = {
+                "since": (dt.date.today() - dt.timedelta(days=META_ADS_MAX_HISTORY_DAYS)).strftime("%Y-%m-%d"),
+                "until": dt.date.today().strftime("%Y-%m-%d"),
+            }
+
+        # Get data based on resource type
+        if is_stats:
+            yield from _get_insights_data(config.account_id, schema, time_range, access_token)
+        else:
+            yield from _get_resource_data(config.account_id, schema, config.resource_name, access_token)
+
+    return SourceResponse(
+        name=name,
+        items=get_rows(),
+        primary_keys=schema.primary_key,
+        partition_count=1,
+        partition_size=1,
+        partition_mode="datetime",
+        partition_format="month",  # the api is daily so bundle into months
+        partition_keys=["date_start"] if is_stats else ["created_time"],
+    )
+
+
+def _get_insights_data(
+    account_id: str,
+    schema: MetaAdsSchema,
+    time_range: dict | None,
+    access_token: str,
+) -> collections.abc.Generator[list[dict], None, None]:
+    """Get insights data from Meta Ads API."""
+
+    params = {
+        "fields": ",".join(schema.field_names),
+        "level": schema.name,
+        "time_increment": 1,
+        "limit": 100,
+    }
+
+    try:
+        url = f"https://graph.facebook.com/{API_VERSION}/{account_id}/insights"
+        rows = []
+
+        for page_data in _make_paginated_api_request(url, params, access_token, time_range):
+            for insight in page_data.get("data", []):
+                row_data = {}
+                for k, v in insight.items():
+                    row_data[k] = _serialize_value(v)
+
+                rows.append(row_data)
+
+                if len(rows) >= DEFAULT_CHUNK_SIZE:
```

Comment:
We dont need to be handling chunk size here. Just yield and let the pipeline take care of this - the logic is already implemented